jobs/cronjob-ldap-group-sync.yml

```diff
@@ -7,44 +7,64 @@ metadata:
     description: "Scheduled Task to Perform LDAP Group Synchronization"
     iconClass: "icon-shadowman"
     tags: "management,cronjob,ldap,group,sync"
+labels:
+  template: "cronjob-ldap-group-sync"
+
 objects:
 - kind: ConfigMap
   apiVersion: v1
   metadata:
     name: ldap-config
+    namespace: "${NAMESPACE}"
     labels:
       template: "cronjob-ldap-group-sync"
   data:
-    ldap-sync.yml: |
-      kind: LDAPSyncConfig
-      apiVersion: v1
-      url: ${LDAP_URL}
-      insecure: true
-      bindDN: ${LDAP_BIND_DN}
-      bindPassword: ${LDAP_BIND_PASSWORD}
+    ldap-group-sync.yaml: |
+      kind: "LDAPSyncConfig"
+      apiVersion: "v1"
+      url: "${LDAP_URL}"
+      insecure: false
+      bindDN: "${LDAP_BIND_DN}"
+      bindPassword:
+        file: "/etc/secrets/bind_password"
+      ca: "/etc/config/ldap-ca.crt"
       rfc2307:
-          groupsQuery:
-              baseDN: ${LDAP_GROUPS_SEARCH_BASE}
-              scope: sub
-              derefAliases: never
-              filter: ${LDAP_GROUPS_FILTER}
-              pageSize: 0
-          groupUIDAttribute: ipaUniqueID
-          groupNameAttributes: [ cn ]
-          groupMembershipAttributes: [ member ]
-          usersQuery:
-              baseDN: ${LDAP_USERS_SEARCH_BASE}
-              scope: sub
-              derefAliases: never
-              pageSize: 0
-          userUIDAttribute: dn
-          userNameAttributes: [ uid ]
-          tolerateMemberNotFoundErrors: true
-          tolerateMemberOutOfScopeErrors: true
+        groupsQuery:
+          baseDN: "${LDAP_GROUPS_SEARCH_BASE}"
+          scope: "sub"
+          derefAliases: "never"
+          filter: "${LDAP_GROUPS_FILTER}"
+          pageSize: 0
+        groupUIDAttribute: "${LDAP_GROUP_UID_ATTRIBUTE}"
+        groupNameAttributes: ${LDAP_GROUP_NAME_ATTRIBUTES}
+        groupMembershipAttributes: ${LDAP_GROUP_MEMBERSHIP_ATTRIBUTES}
+        usersQuery:
+          baseDN: "${LDAP_USERS_SEARCH_BASE}"
+          scope: "sub"
+          derefAliases: "never"
+          pageSize: 0
+        userNameAttributes: ${LDAP_USER_NAME_ATTRIBUTES}
+        userUIDAttribute: "${LDAP_USER_UID_ATTRIBUTE}"
+        tolerateMemberNotFoundErrors: true
+        tolerateMemberOutOfScopeErrors: true
+    ldap-ca.crt: "${LDAP_CA_CERT}"
```

Comment:
Obviously, `ldaps` should be preferred over just `ldap`, but what if `ldap` is used and not `ldaps`? this template now requires it to be `ldaps` (and the user HAS to supply a CA even if `ldap` is used`) Could this value just be a "garbage" string if `ldap` is used (instead of `ldap`)? If so, please default a value in the parameters to avoid requiring this to be set for cases where `ldap` is used

Reply:
I guess I regard using ldap without TLS for any aspect of cluster security as so wildly irresponsible that it would be better to break and make people do some soul-searching about why they don't have at the very least a self-signed certificate.

I have added parameters LDAP_INSECURE and LDAP_CA_CERT_PATH which can be used to disable TLS by setting LDAP_INSECURE=true and LDAP_CA_CERT_PATH=''